Music metadata

roipoussiere opened the pull request:
As explained in #578, this adds support for music metadata.

- add function to look for music metadata by looking for tags in code comments;
- update some tunes to add those metadata;
- display title from music metadata in user patterns;
- display title from music metadata in examples page;
- add metadata section in the docs.

felixroos commented:
great, thanks so much!

felixroos commented on website/src/pages/learn/code.mdx:
```diff
@@ -67,6 +67,9 @@ It is a handy way to quickly turn code on and off.
 You can also use the keyboard shortcut `cmd-/` to toggle comments on and off.
 
+You might noticed that some comments in the REPL samples include some words starting with a "@", like `@title` or `@license`.
+Those are just a convention to define some information about the music. We will talk about it in the _metadata_ section.
```
makes sense to link the metadata page here

felixroos commented on website/src/pages/learn/metadata.mdx:
````diff
@@ -0,0 +1,31 @@
+You can optionally add some music metadata in your Strudel code, by using tags in code comments like this:
+
+```
+// @license: CC BY-NC-SA 4.0 https://creativecommons.org/licenses/by-nc-sa/4.0/
````
I wonder why it's with colons here, but the examples don't use colons?

roipoussiere replied:
Just a mistake here, I will remove it.
I made colons optional, so the music title should be the same anyway.

felixroos commented on website/src/pages/learn/metadata.mdx:
```diff
@@ -0,0 +1,31 @@
+Available tags are:
+
+- `@title`: music title
+- `@by`: music author(s), separated with comma, eventually followed with a link in `<>` (ex: `@author john doe <https://example.com>`)
```
typo: example still uses @author

roipoussiere commented:
Thanks for the review.

Updates:
- parser now supports multiple values and one-liners (see doc);
- added tests;
- fixed and improved documentation.

felixroos commented on website/src/pages/learn/metadata.mdx:
````diff
@@ -0,0 +1,88 @@
+
+```js
+/*
+song @by Sam Tagada
````
thats pretty cool

roipoussiere replied:
Thank! It's even possible to define multiple licenses as well:

```
// song @by Sam @license CC0
// samples @by Jimmy @license CC BY-NC-SA
```

felixroos commented on website/src/pages/learn/metadata.mdx:
````diff
@@ -0,0 +1,88 @@
+*/
+```
+
+You can also add optinal prefixes and use tags where you want:
````
typo: optinal

roipoussiere commented:
I find the syntax `// @title Awesome song @by Sam` a bit too verbose and disturbing, so I added an alternative syntax using quotes: `// "Awesome song" @by Sam`.

roipoussiere commented:
Ready to merge for me. :)
